### PR TITLE
Fix link tabs not showing selected tab with nested routes

### PR DESCRIPTION
**Pull request by iamvukasin**

### CODE REVIEW

File: packages/tempus-client_v3/src/components/App/App.tsx [hunk 1]
```diff
@@ -21,8 +21,16 @@ const App = () => {
   const { t } = useTranslation();
 
   const navigationLinks: PageNavigationLink[] = [
-    { text: t('App.navMarkets'), path: '/' },
-    { text: t('App.navPortfolio'), path: '/portfolio/overview' },
+    {
+      text: t('App.navMarkets'),
+      path: '/',
+      pathPatterns: ['/', '/pool/*'],
```
Comment (ShawTim) [inline packages/tempus-client_v3/src/components/App/App.tsx:27]: so if we have `pathPatterns`, we wont use `path`?

File: packages/tempus-client_v3/src/components/shared/Tabs/Tabs.tsx [hunk 2]
```diff
@@ -15,23 +15,34 @@ export interface TabsProps {
...
-  const values = Children.map(children, child => child?.props.href ?? child?.props.value);
-  const selectedTabIndex = useMemo(
-    () => (values ? values.findIndex(tabValue => tabValue === value) : 0),
-    [value, values],
-  );
 
-  const tabWidth = useMemo(() => 100 / (values ? values.length : 1), [values]);
+  const selectedTabIndex = useMemo(() => {
+    const selectedIndex = Children.map(children, child => {
+      if (child?.props.hrefPatterns) {
```
Comment (ShawTim) [inline packages/tempus-client_v3/src/components/shared/Tabs/Tabs.tsx:21]: move the question to here.
so apparently if the `props` contain `hrefPatterns`, `href` will not be used. am i correct?

Comment (iamvukasin) [reply packages/tempus-client_v3/src/components/shared/Tabs/Tabs.tsx:21]: Yes. In that case, `hrefPatterns` has priority when checking `location.pathname` match. `href` would still be used inside `Tab` for links.

Comment (iamvukasin) [reply packages/tempus-client_v3/src/components/shared/Tabs/Tabs.tsx:21]: Not having `hrefPatterns` effectively means that you have only match when `location.pathname === href`.

Comment (ShawTim) [reply packages/tempus-client_v3/src/components/shared/Tabs/Tabs.tsx:21]: in that case then it actually not necessary to pass `href` (which is `path` in `App.tsx`) as param if `pathPatterns` is provided

Comment (ShawTim) [reply packages/tempus-client_v3/src/components/shared/Tabs/Tabs.tsx:21]: or, is it possible to accept a `paths` params, which is `string | (string | Regexp)[]`?

Comment (iamvukasin) [reply packages/tempus-client_v3/src/components/shared/Tabs/Tabs.tsx:21]: `path` is for link itself, but `pathPatterns` is used to check if the current URL pathname matches any of `pathPatterns`. We have to keep them both.

**Review by ShawTim — APPROVED** [review]

**Pull request merged**